daft/runners/partitioning.py

```diff
@@ -126,7 +156,7 @@ def get_partition(self, idx: PartID) -> PartitionT:
         raise NotImplementedError()
 
     @abstractmethod
-    def set_partition(self, idx: PartID, part: PartitionT) -> None:
+    def set_partition(self, idx: PartID, part: MaterializedResult[PartitionT]) -> None:
```

Comment:
`set_partition` now takes as input a `MaterializedResult` instead of a plain old `PartitionT`.